textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/neextrator/NamedEntityExtractor.java

```diff
@@ -72,7 +85,168 @@ public void open() throws Exception {
+     * its NamedEntityAnnotation, if it's a valid value (not 'O'),
+     * then makes it a span and add to the return list.
+     * The Stanford NLP constants are mapped into the NE constants.
+     * The NLP package has annotations for the start and end position of a token
+     * and it perfectly matches the span design so we just use them.
+     *
+     */
+    private List<Span> extractNESpans(IField iField, String fieldName) {
+        List<Span> spanList = new ArrayList<>();
+        String text = (String) iField.getValue();
+        Properties props = new Properties();
+        props.setProperty("annotators", "tokenize, ssplit, pos, lemma, ner");
+        StanfordCoreNLP pipeline = new StanfordCoreNLP(props);
+        Annotation documentAnnotation = new Annotation(text);
+        pipeline.annotate(documentAnnotation);
+        List<CoreMap> sentences = documentAnnotation.get(CoreAnnotations.SentencesAnnotation.class);
+        for (CoreMap sentence : sentences) {
+            for (CoreLabel token : sentence.get(CoreAnnotations.TokensAnnotation.class)) {
+                String NLPConstant = token.get(CoreAnnotations.NamedEntityTagAnnotation.class);
+                if (!NLPConstant.equals("O")) {
+
+                    String NEConstant = getNEConstant(NLPConstant);
+                    int start = token.get(CoreAnnotations.CharacterOffsetBeginAnnotation.class);
+                    int end = token.get(CoreAnnotations.CharacterOffsetEndAnnotation.class);
+                    String word = token.get(CoreAnnotations.TextAnnotation.class);
+
+                    Span span = new Span(fieldName, start, end, NEConstant, word);
+
+
+                    if (spanList.size() >= 1) {
+                        Span previousSpan = spanList.get(spanList.size() - 1);
+                        if (previousSpan.getFieldName().equals(span.getFieldName())
+                                && (span.getStart() - previousSpan.getEnd() <= 1)
+                                && previousSpan.getKey().equals(span.getKey())) {
+                            Span newSpan = mergeTwoSpans(previousSpan, span);
+                            span = newSpan;
+                            spanList.remove(spanList.size() - 1);
+                        }
+                    }
+                    spanList.add(span);
+
+                }
+            }
+
+        }
+
+        return spanList;
+    }
+
+
+    /**
+     * @param previousSpan
+     * @param currentSpan
+     * @return
+     * @about This function takes two spans as input and merges them as a new span
+     * <p>
+     * Two spans with fieldName, start, end, key, value:
+     * previousSpan: "Doc1", 10, 13, "Location", "New"
+     * currentSpan : "Doc1", 14, 18, "Location", "York"
+     * <p>
+     * Would be merge to:
+     * return:   "Doc1", 10, 18, "Location", "New York"
+     * <p>
+     * The caller needs to make sure:
+     * 1. The two spans are adjacent.
```

Comment:
@chenlica I don't want to add this inside this method. It's a private method only being used inside this class and condition should satisfy before calling this method. The condition checking part is in line156 ~line 158. If this looks good I'll make the merge.